plugins/modules/zos_zfs_resize.py

```diff
@@ -0,0 +1,567 @@
+        module.fail_json(
+            msg='Parameter verification failed.',
+            stderr=str(err)
+        )
+
+    result = dict()
+    target = module.params.get("target")
+    size = module.params.get("size")
+    space_type = module.params.get("space_type")
+    noai = module.params.get("no_auto_increment")
+    verbose = module.params.get("verbose")
+    trace_destination = module.params.get("trace_destination")
+
+    changed = False
+    # Variables to return the value on the space_type by the user
+    size_on_type = ""
+    free_on_type = ""
+
+    result.update(
+        dict(
+            target=target,
+            size=size,
+            space_type=space_type.upper(),
+            cmd="",
+            changed=changed,
+            rc=1,
+            stdout="",
+            stderr="",
+        )
+    )
+
+    if not (data_set.DataSet.data_set_exists(target)):
+        module.fail_json(msg="ZFS Target {0} does not exist".format(target), **result)
+
+    # Validation to found target on the system and also get the mount_point
+    mount_target = find_mount_target(module=module, target=target, results=result)
+
+    # Initialize the class with the target
+    zfsadm_obj = zfsadm(aggregate_name=target, module=module)
+
+    rc, stdout, stderr = zfsadm.get_aggregate_size(zfsadm_obj.aggregate_name, module)
+
+    if rc == 0:
+        old_size, old_free = get_size_and_free(line=stdout)
+
+    if space_type != "k":
+        space = calculate_size_on_k(size=size, space_type=space_type)
+    else:
+        space = size
+
+    # Validations to know witch function will be execute
+    operation = ""
+    minimum_size_t_shrink = old_size - old_free
+
+    if space_type != "k":
+        size_on_type = convert_size(size=old_size, space_type=space_type)
+        free_on_type = convert_size(size=old_size, space_type=space_type)
+
+    str_old_size = old_size if space_type == "k" else size_on_type
+    str_old_size = float("{:.1f}".format(str_old_size))
+    str_old_free = old_free if space_type == "k" else free_on_type
+    str_old_free = float("{:.1f}".format(str_old_free))
+
+    result.update(
+        dict(
+            mount_target=mount_target,
+            old_size=str_old_size,
+            old_free=str_old_free,
+        )
+    )
+
+    if space == old_size:
+        result.update(
+            dict(
+                cmd="",
+                rc=0,
+                stdout="Same size as size of the file {0}".format(target),
+                stderr="",
+                changed=False,
+                size=size,
+                new_size=str_old_size,
+                new_free=str_old_free,
+            )
+        )
+        module.exit_json(**result)
+
+    elif space > old_size:
+        operation = "grow"
+
+    elif space >= minimum_size_t_shrink and space < old_size:
+        operation = "shrink"
+
+    elif space < minimum_size_t_shrink:
+        module.fail_json(msg="Not enough space to shrink", **result)
+
```

Comment:
Here is the zinger: if you pass a '0' for size, it returns an error (0 < minimum size of shrink).  I was trying to figure out a way to do it without failure, like passing a 'no_operation' flag to just get the size and return it in the structure.